dotnet/samples/04.ai.e.chainedActions.devOpsBot/Program.cs

```diff
@@ -0,0 +1,153 @@
+﻿using DevOpsBot;
+using DevOpsBot.Model;
+
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Integration.AspNet.Core;
+using Microsoft.Bot.Connector.Authentication;
+using Microsoft.TeamsAI;
+using Microsoft.TeamsAI.AI;
+using Microsoft.TeamsAI.AI.Moderator;
+using Microsoft.TeamsAI.AI.Planner;
+using Microsoft.TeamsAI.AI.Prompt;
+
+WebApplicationBuilder builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddControllers();
+builder.Services.AddHttpClient("WebClient", client => client.Timeout = TimeSpan.FromSeconds(600));
+builder.Services.AddHttpContextAccessor();
+
+// Prepare Configuration for ConfigurationBotFrameworkAuthentication
+ConfigOptions config = builder.Configuration.Get<ConfigOptions>()!;
+builder.Configuration["MicrosoftAppType"] = "MultiTenant";
+builder.Configuration["MicrosoftAppId"] = config.BOT_ID;
+builder.Configuration["MicrosoftAppPassword"] = config.BOT_PASSWORD;
+
+// Create the Bot Framework Authentication to be used with the Bot Adapter.
+builder.Services.AddSingleton<BotFrameworkAuthentication, ConfigurationBotFrameworkAuthentication>();
+
+// Create the Cloud Adapter with error handling enabled.
+// Note: some classes expect a BotAdapter and some expect a BotFrameworkHttpAdapter, so
+// register the same adapter instance for all types.
+builder.Services.AddSingleton<CloudAdapter, AdapterWithErrorHandler>();
+builder.Services.AddSingleton<IBotFrameworkHttpAdapter>(sp => sp.GetService<CloudAdapter>()!);
+builder.Services.AddSingleton<BotAdapter>(sp => sp.GetService<CloudAdapter>()!);
+
+// Create singleton instances for bot application
+builder.Services.AddSingleton<IStorage, MemoryStorage>();
+
+#region Use OpenAI
+// Use OpenAI
+if (config.OpenAI == null || string.IsNullOrEmpty(config.OpenAI.ApiKey))
+{
+    throw new ArgumentException("Missing OpenAI configuration.");
+}
+builder.Services.AddSingleton<OpenAIPlannerOptions>(_ => new(config.OpenAI.ApiKey, "gpt-3.5-turbo"));
+builder.Services.AddSingleton<OpenAIModeratorOptions>(_ => new(config.OpenAI.ApiKey, ModerationType.Both));
```

Comment:
The js sample does not use moderator. Better to keep consistent for simplicity.

Reply:
Good point. Removed moderator.
I'll check this on other samples as well.